sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/TryCastSuite.scala

```diff
@@ -79,3 +99,14 @@ class TryCastSuite extends SparkFunSuite {
     assert(!c4.resolved)
   }
 }
+
+class TryCastThrowExceptionSuite extends SparkFunSuite with ExpressionEvalHelper {
+  // The method checkExceptionInExpression is overridedn in TryCastSuite, so here we have a
+  // new test suite for testing exceptions from the child of `try_cast()`.
+  test("TryCast should not catch the exception from it's child") {
```

Comment:
why can't we put this test in `TryCastSuite`?

Reply:
Please read my comment above. 
The method checkExceptionInExpression is overridden in TryCastSuite for testing all the cases should return nulls.